engine/api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/CustomAction.scala

```diff
@@ -0,0 +1,29 @@
+package pl.touk.nussknacker.engine.api.deployment
+
+import pl.touk.nussknacker.engine.api.process.ProcessName
+
+import java.net.URI
+
+/*
+This is an experimental/wip version.
+FIXME:
+1. There is no validation on BE to check if given action can be processed,
+   eg if an action is some sort of custom process deployment we should validate it against current process status.
+   For now we only disable custom action buttons on FE when action's allowed process states doesn't include current process state.
+
+Things to consider in future changes:
+1. Allowing for attaching comments to custom actions, similarly to stop/deploy comments.
+2. Storing custom actions taken in DB.
+ */
+
+case class CustomAction(name: String,
+                        allowedProcessStates: List[StateStatus],
+                        icon: Option[URI] = None)
+
+case class CustomActionRequest(name: String,
```

Comment:
I wonder, will this be enough, e.g. shall we also have access to process?
I think we won't be able e.g. to invoke Flink job with these information?
We can also e.g. pass user invoking action?

Reply:
Isn't that to invoke flink job we additionally need savepoint path, which can be passed as parameter?

Reply:
@mproch